src/js/app.js

```diff
@@ -0,0 +1,42 @@
+import { $ } from "./utils.js";
+import { calculator } from "./calculator.js";
+
+class App {
+  total = $("#total");
+
+  init() {
+    this.setEventListener();
+  }
+
+  setEventListener() {
+    const calculatorArea = $(".calculator");
+
+    calculatorArea.addEventListener("click", (e) => {
+      if (e.target.className === "digit") {
+        const digit = e.target.innerText;
+        calculator.pressDigit(digit);
+        this.showNumber(calculator.current);
+        return;
+      }
+
+      if (e.target.className === "operation") {
+        const operator = e.target.innerText;
+        calculator.pressOperator(operator);
+        this.showNumber(calculator.result);
+
+        return;
+      }
+
+      if (e.target.className === "modifier") {
+        calculator.allClear();
+        this.showNumber(calculator.result);
+      }
```

Comment:
`this.showNumber` 메서드가 모든 케이스에서 실행되는 것 같아서 개선해 볼 수 있을거같아요~

`e.target.className`에 따라서 calculator가 실행하는 동작들을 메서드로 분리해서 실행시키고 아래에 showNumber을 해보면 어떨까요?
```suggestion
      if (e.target.className === "digit") {
        const digit = e.target.innerText;
        calculator.pressDigit(digit);
        this.showNumber(calculator.current);
        return;
      }

      if (e.target.className === "operation") {
        const operator = e.target.innerText;
        calculator.pressOperator(operator);
        this.showNumber(calculator.result);

        return;
      }
handleClick~()
this.showNumber(calculator[??]);
```
실행해야할 calculator의 메서드의 key를 Lookup table을 활용해서 가져오면 이렇게 바꿔볼 수 있을 것 같아요 🤔

Reply:
Lookup table 활용이 어려운데 의도적으로라도 자주 사용해야겠습니다 😰